third_party/move/move-vm/runtime/src/loader.rs

```diff
@@ -745,11 +742,128 @@ impl Loader {
             .collect::<PartialVMResult<Vec<_>>>()
             .map_err(|err| err.finish(Location::Undefined))?;
 
-        // verify type arguments
+        Ok((module, func, parameters, return_))
+    }
+
+    // Matches the actual returned type to the expected type, binding any type args to the
+    // necessary type as stored in the map. Returns true if a successful match is made.
+    fn match_return_type<'a>(
+        returned: &Type,
+        expected: &'a Type,
+        map: &mut BTreeMap<usize, &'a Type>,
+    ) -> bool {
+        match (returned, expected) {
+            // The important case, deduce the type params
+            (Type::TyParam(idx), _) => {
+                if let Option::Some(bounded_type) = map.get(idx) {
```

Comment:
Nit: take a look at the [entry API](https://doc.rust-lang.org/stable/std/collections/btree_map/struct.BTreeMap.html#method.entry). It can help you save a lookup in cases like this.

Reply:
Mentioned this privately, but you can match against the two variants of Entry: https://doc.rust-lang.org/stable/std/collections/btree_map/enum.Entry.html

Reply:
Done